### PR TITLE
Bug 1194767 - use 'nice' slugids from slugid.py package

### PR DESCRIPTION
**Pull request by petemoore**
Regarding the tests, now that the new slugid library has [its own tests](https://github.com/taskcluster/slugid.py/blob/master/test.py), there is an argument we could remove the slug tests from this library. However maybe it is also nice to keep them as an extra safety guarantee. The current tests in this library stub the `uuid.uuid4()` function since previously the slugid generation process in this taskcluster client was using `uuid.uuid4()`. The new [slugid client](https://pypi.python.org/pypi/slugid) is also using `uuid.uuid4()`, but since now this is implemented in an external library, this could change, so if we keep the tests we also now making assumptions about the other library's implementation. This might be another reason to remove the tests.

I don't mind either way, let me know what you think.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,7 +17,8 @@
   'subprocess32==3.2.6',
   'psutil==2.1.3',
   'hypothesis',
-  'pgpy'
+  'pgpy',
+  'slugid'
```
Comment (rail) [inline setup.py]: Shouldn't this go into install_requires?

File: taskcluster/utils.py [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import re
 import json
 import datetime
-import uuid
 import base64
 import logging
 import os
 import requests
+import slugid as slugid_
```
Comment (rail) [inline taskcluster/utils.py]: I don't think you need to use "as slugid_" here. It doesn't collide with other variables. Just wanted it have it more distinguishable from slugId()?

**Comment by rail** [comment]
Otherwise lgtm!

File: taskcluster/utils.py (hunk 2 above)
Comment (petemoore) [inline taskcluster/utils.py]: Ah yes, my mistake! The function defined below is `slugId` not `slugid` so there was no collision. Good spot. :+1:

File: setup.py (hunk 1 above)
Comment (petemoore) [inline setup.py]: Whoooooooooooops :grimacing:

**Comment by jhford** [comment]
Looks good to me!  feel free to merge whenever you like

**Pull request merged**